### PR TITLE
Remove Base64 logic from embedded resources

### PR DESCRIPTION
**Pull request by JD557**
Removes the Base64 encoding/decoding logic when embedding resources.

I believe there was a misunderstanding in #2475 regarding the original idea proposed in #567

The [original message](https://github.com/scala-native/scala-native/issues/567#issuecomment-380032447) (emphasis mine)

>An easy way to support the underlying use case would be to add magical `base64"...": Ptr[Byte]` constant. The value of constant would be stored in the read-only section of the binary **(i.e. not encoded as base64, but raw data)**.
>End users would use a code generation step to convert all of the resources into base64 as part of their build.

So I think the base64 part was supposed to be an intermediate step just to simplify the codegen, but in reality that would never end in the final binary.

### CODE REVIEW

File: scripted-tests/run/java-net-socket/SocketHelpers.scala [hunk 1]
```diff
@@ -0,0 +1,356 @@
+package java.net
```
[the remark this answers is not in the document]
Comment (JD557) [reply scripted-tests/run/java-net-socket/SocketHelpers.scala:1]: No idea... I'll clean this up and do a push force later... I must have messed something

[the remark this answers is not in the document]
Comment (JD557) [reply scripted-tests/run/java-net-socket/SocketHelpers.scala]: Aha, that file is created when you run the scripted tests https://github.com/scala-native/scala-native/blob/main/project/Build.scala#L144-L153

**Comment by JD557** [comment]
The build is failing because I renamed `EncodedResourceInputStream` to `EmbeddedResourceInputStream` (as there is no encoding going on anymore).

This appears to be a safe change, since that class is marked as `private[lang]`.

**Comment by armanbilge** [comment]
Are you familiar with add MiMa binary issue filters? Haven't checked the build to see where they should go but I believe that should do the trick.

**Comment by JD557** [comment]
I was just leaving a comment for "future me", but I'll take a look, thanks.

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**